commit 48c8a116a914a325a0497721f5d8b58d5bba34d4
Author: Paul Smith <psmith@gnu.org>
Date:   Sun Nov 19 15:09:16 2017 -0500

    * configure.ac: Support GLIBC glob interface version 2

commit 193f1e81edd6b1b56b0eb0ff8aa4b41c7b4257b4
Author: Paul Eggert <eggert@cs.ucla.edu>
Date:   Sun Sep 24 09:12:58 2017 -0400

    glob: Do not assume glibc glob internals.
    
    It has been proposed that glibc glob start using gl_lstat,
    which the API allows it to do.  GNU 'make' should not get in
    the way of this.  See:
    https://sourceware.org/ml/libc-alpha/2017-09/msg00409.html
    
    * dir.c (local_lstat): New function, like local_stat.
    (dir_setup_glob): Use it to initialize gl_lstat too, as the API
    requires.

--- gridengine.orig/source/3rdparty/qmake/configure.in
+++ gridengine/source/3rdparty/qmake/configure.in
@@ -353,10 +353,9 @@
 #include <glob.h>
 #include <fnmatch.h>
 
-#define GLOB_INTERFACE_VERSION 1
 #if !defined _LIBC && defined __GNU_LIBRARY__ && __GNU_LIBRARY__ > 1
 # include <gnu-versions.h>
-# if _GNU_GLOB_INTERFACE_VERSION == GLOB_INTERFACE_VERSION
+# if _GNU_GLOB_INTERFACE_VERSION == 1 || _GNU_GLOB_INTERFACE_VERSION == 2
    gnu glob
 # endif
 #endif
--- gridengine.orig/source/3rdparty/qmake/dir.c
+++ gridengine/source/3rdparty/qmake/dir.c
@@ -1206,15 +1206,40 @@
 }
 #endif
 
+/* Similarly for lstat.  */
+#if !defined(lstat) && !defined(WINDOWS32) || defined(VMS)
+# ifndef VMS
+#  ifndef HAVE_SYS_STAT_H
+int lstat (const char *path, struct stat *sbuf);
+#  endif
+# else
+    /* We are done with the fake lstat.  Go back to the real lstat */
+#   ifdef lstat
+#     undef lstat
+#   endif
+# endif
+# define local_lstat lstat
+#elif defined(WINDOWS32)
+/* Windows doesn't support lstat().  */
+# define local_lstat local_stat
+#else
+static int
+local_lstat (const char *path, struct stat *buf)
+{
+  int e;
+  EINTRLOOP (e, lstat (path, buf));
+  return e;
+}
+#endif
+
 void
 dir_setup_glob (glob_t *gl)
 {
   gl->gl_opendir = open_dirstream;
   gl->gl_readdir = read_dirstream;
   gl->gl_closedir = ansi_free;
+  gl->gl_lstat = local_lstat;
   gl->gl_stat = local_stat;
-  /* We don't bother setting gl_lstat, since glob never calls it.
-     The slot is only there for compatibility with 4.4 BSD.  */
 }
 
 void
